importanize/__main__.py

```diff
@@ -1,396 +1,10 @@
-            if not f.is_file() or f.is_file() and f.suffixes == [".py"]
-        )
-
-        all_successes = True
-        for f in files:
-            try:
-                for i in run(f, config, args, f):
-                    yield i
-            except CIFailure:
-                all_successes = False
-
-        if not all_successes:
-            raise CIFailure()
-
-
-def is_piped():
-    return S_ISFIFO(os.fstat(0).st_mode)
-
-
-def main(args=None):
-    args = args if args is not None else sys.argv[1:]
-    args = parser.parse_args(args=args)
-    # adjust logging level
-    (logging.getLogger("").setLevel(VERBOSITY_MAPPING.get(args.verbose, 0)))
-
-    log.debug("Running importanize with {}".format(args))
-
-    if args.py and args.py != sys.version_info.major:
-        log.debug(
-            "Exiting as running in Python {} but only allowed to run in Python {}"
-            "".format(sys.version_info.major, args.py)
-        )
-        return 0
-
-    config = Config.from_path(getattr(args.config, "name", "")) or Config.find()
-
-    if args.version:
-        msg = (
-            "importanize\n"
-            "===========\n"
-            "{description}\n\n"
-            "version: {version}\n"
-            "python: {python}\n"
-            "source: https://github.com/miki725/importanize\n\n"
-            "root config ({config}):\n"
-            "{all_config}"
-        )
-        print(
-            msg.format(
-                description=__description__,
-                version=__version__,
-                python=sys.executable,
-                config=config,
-                all_config=pprint.pformat(config),
-            )
-        )
-        return 0
-
-    to_importanize = [pathlib.Path(i) for i in (args.path or ["."])]
-
-    if is_piped() and not args.path:
-        to_importanize = [force_text(sys.stdin.read())]
-        args.print = True
-        args.header = False
-
-    if args.ci or args.list:
-        args.print = False
-        args.header = False
-
-    all_successes = True
-    all_groups = []
-
-    for p in to_importanize:
-        try:
-            all_groups += [i for i in run(p, config, args)]
-        except CIFailure:
-            all_successes = False
-        except Exception:
-            log.exception("Error running importanize")
-            return 1
-
-    if args.list:
-        groups = ImportGroups()
-        for c in config["groups"]:
-            groups.add_group(c)
-        statements = chain(*(i.statements for i in chain(*all_groups)))
-        for s in statements:
-            groups.add_statement_to_group(s)
-        for g in groups:
-            print(g.config["type"])
-            print("-" * len(g.config["type"]))
-            for s in g.unique_statements:
-                print(six.text_type(s))
-            print()
-
-    return int(not all_successes)
-
-
-sys.exit(main()) if __name__ == "__main__" else None
+click_cli() if __name__ == "__main__" else None
```

Comment:
```suggestion
if __name__ == "__main__":
    click_cli()
```

The more elegant way.

Reply:
true but in that case line 11 is not coverable. it could be ignore with a comment but that feels hacky too.